resolwe/flow/models/data.py

```diff
@@ -230,6 +230,17 @@ def duplicate(
 
         return duplicated
 
+    @transaction.atomic
+    def move_to_collection(self, destination_collection):
+        """Move data objects to destination collection."""
+        for datum in self:
+            datum.validate_collection(destination_collection)
+            datum.collection = destination_collection
+            if destination_collection:
+                datum.tags = destination_collection.tags
+                copy_permissions(destination_collection, datum)
```

Comment:
I've taken this from `Entity` model but noticed that if you change data collection by updating collection on data directly via 

```
request = {"collection": {"id": self.collection.pk}}
data_detail_viewset(request, pk=data.pk)
```

then I guess tags and permissions aren't copied. Should tags and permissions also be copied in the latter case?

Reply:
I strongly vote for yes. As an direct restAPI user i would like that such intricacies are solved "for free". I guess this also removes some of the burden from frontend, right? Another use case is ReSDK, i'm sure users will be happy if they are not required to copy collection tags and permissions manually. 
Drawback is of course the lost of general ability to do one isolated "move". But do we need it anyway? Is there such a use case?

Reply:
@dblenkus what's your take on this?

Reply:
Or we can leave as it is and use `move_to_collection` to when we want to copy tags and permissions. You'll just put a single data object in the input list.

Reply:
I think that it is necessary to keep collection, entity, and data in sync. So it would be great if you can fix that.
Regarding backward compatibility, I think that it is better to remove/change it than to leave an option to put things out of sync.

Reply:
Like [this](https://github.com/genialis/resolwe/pull/711/commits/5932e9a6deda3bc1371baea42867312045ce52bc)?